src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientHandler.java

```diff
@@ -50,34 +51,48 @@ public static boolean isClientValid(SnowflakeStreamingIngestClient client) {
     return client != null && !client.isClosed() && client.getName() != null;
   }
 
+  /**
+   * Gets the Properties from the input connector config
+   *
+   * @param connectorConfig configuration properties for a connector
+   * @return the Properties object needed for client creation
+   */
+  public static Properties getClientProperties(Map<String, String> connectorConfig) {
+    Properties streamingClientProps = new Properties();
+
+    if (connectorConfig != null) {
+      streamingClientProps.putAll(
+          StreamingUtils.convertConfigForStreamingClient(new HashMap<>(connectorConfig)));
+    }
+
+    return streamingClientProps;
+  }
+
   /**
    * Creates a streaming client from the given config
    *
    * @param connectorConfig The config to create the client
-   * @return A newly created client
+   * @return The client properties and the newly created client
    */
-  public SnowflakeStreamingIngestClient createClient(Map<String, String> connectorConfig) {
+  public Pair<Properties, SnowflakeStreamingIngestClient> createClient(
+      Map<String, String> connectorConfig) {
     LOGGER.info("Initializing Streaming Client...");
 
-    // get streaming properties from config
-    Properties streamingClientProps = new Properties();
-    streamingClientProps.putAll(
-        StreamingUtils.convertConfigForStreamingClient(new HashMap<>(connectorConfig)));
+    Properties streamingClientProps = getClientProperties(connectorConfig);
 
-    try {
-      // Override only if bdec version is explicitly set in config, default to the version set
-      // inside Ingest SDK
-      Map<String, Object> parameterOverrides = new HashMap<>();
-      Optional<String> snowpipeStreamingBdecVersion =
-          Optional.ofNullable(connectorConfig.get(SNOWPIPE_STREAMING_FILE_VERSION));
-      snowpipeStreamingBdecVersion.ifPresent(
-          overriddenValue -> {
-            LOGGER.info("Config is overridden for {} ", SNOWPIPE_STREAMING_FILE_VERSION);
-            parameterOverrides.put(BLOB_FORMAT_VERSION, overriddenValue);
-          });
+    // Override only if bdec version is explicitly set in config, default to the version set
+    // inside Ingest SDK
+    Map<String, Object> parameterOverrides = new HashMap<>();
```

Comment:
moved otuside of the try-catch because we don't need to catch any exceptions that come from this

Reply:
I don't think it matters much, but why we don't want to catch the exception from this line?

Reply:
yeah it doesn't matter too much. I moved it out because I prefer to minimize scope of the try catch